xarray/core/concat.py

```diff
@@ -668,6 +681,7 @@ def get_indexes(name):
         result_indexes[dim] = index
 
     # TODO: add indexes at Dataset creation (when it is supported)
+    # TODO: do we actually need this step anymore?
     result = result._overwrite_indexes(result_indexes)
```

Comment:
Commenting this line out does not break any tests in `test_concat.py`, which implies that it isn't even being used now.

Reply:
Oops replied in https://github.com/pydata/xarray/pull/8872#discussion_r1539106484.